File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    def dfs(v):
        visited[v] = True
        for i in range(n):
            if i == v:
                continue
            if computers[v][i] == 1 and not visited[i]:
                dfs(i)
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[200];

int Dfs(int n, vector<vector<int>>& computers, int v){
    for (int i =0; i < n; i++){
        if (i != n && !visited[i] && computers[v][i]){
            visited[i] = true;
            Dfs(n, computers, i);
        }
    }
    return 0;
}

int solution(int n, vector<vector<int>> computers) {
    int ans = 0;
    for (int i = 0; i < n; i++){
        if (!visited[i]){
            visited[i] = true;
            Dfs(n, computers, i);
            ans++;
        }
    }

    return ans;
}